flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/TableStoreFactory.java

```diff
@@ -84,14 +109,15 @@ public void onCreateTable(Context context, boolean ignoreIfExists) {
                                 context.getObjectIdentifier().asSerializableString()));
             }
             path.getFileSystem().mkdirs(path);
+            options.put(TABLE_PATH.key(), path.getPath());
```

Comment:
1.I think we can enrich table path in the `enrichOptions`.
2.Maybe we can support table path configuration from user.
3.`path.getPath` is wrong, it lacks scheme, we should use `path.toString`.

Reply:
I have some questions here. 

1. If the users specify `table.path`, does it indicate `file.path` is useless and can be simplified as one?

2. Let users specify the `table.path` means we lose control of the relative table path/table name. We cannot reconstruct the relative path/table name simply from `tableIdentifier`. If we give users this choice, do we also consider let users name their topic freely?

Reply:
> 1.I think we can enrich table path in the `enrichOptions`. 2.Maybe we can support table path configuration from user. 3.`path.getPath` is wrong, it lacks scheme, we should use `path.toString`.

I think we can enrich `table.path` in `enrichOptions`, but still keep this option internal

Reply:
I agree we can keep table path internal.
But I think we should remove `TABLE_PATH`. This is weird. We can pass `tableIdentifier` to `FileStoreOptions`.

Reply:
> I agree we can keep table path internal. But I think we should remove `TABLE_PATH`. This is weird. We can pass `tableIdentifier` to `FileStoreOptions`.

Sure. This is to be resolved in https://github.com/apache/flink-table-store/pull/45